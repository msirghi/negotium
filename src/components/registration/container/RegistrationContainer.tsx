import { AuthContainer } from '../../login';
import { RegistrationForm } from '../form/RegistrationForm';
import { RegistrationFooter } from './footer/RegistrationFooter';

export const RegistrationContainer = () => {
  return (
    <AuthContainer footer={RegistrationFooter}>
      <RegistrationForm />
    </AuthContainer>
  );
};
