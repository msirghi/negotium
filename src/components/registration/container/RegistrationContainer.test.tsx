import { mount } from 'enzyme';
import renderer, { act } from 'react-test-renderer';
import { RegistrationContainer } from './RegistrationContainer';
import { SnackbarProvider } from 'notistack';

const mockPush = jest.fn();

jest.mock('next/router', () => ({
  useRouter: () => ({
    route: 'inbox',
    query: {},
    push: mockPush,
  }),
}));

describe('Registration container', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(
      <SnackbarProvider>
        <RegistrationContainer />
      </SnackbarProvider>
    );
    expect(tree).toMatchSnapshot();
  });

  it('should handle sign in click', () => {
    const wrapper = mount(
      <SnackbarProvider>
        <RegistrationContainer />
      </SnackbarProvider>
    );
    act(() => {
      wrapper.find('#sign-in-link').at(0).simulate('click');
    });
    expect(mockPush).toBeCalled();
  });
});
