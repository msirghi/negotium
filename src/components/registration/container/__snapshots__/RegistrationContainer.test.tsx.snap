// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration container should match the snapshot 1`] = `
<div
  className="makeStyles-container-3"
>
  <div
    className="makeStyles-contentCard-4"
  >
    <div
      className="makeStyles-title-5"
    >
      common.welcome
    </div>
    <div
      className="makeStyles-iconContainer-7"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium makeStyles-icon-8 css-ci5apu-MuiSvgIcon-root"
        data-testid="AcUnitIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22z"
        />
      </svg>
    </div>
    <div
      className="makeStyles-loginForm-6"
    >
      <form
        onSubmit={[Function]}
      >
        
        <div>
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            style={
              Object {
                "marginTop": 30,
              }
            }
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
            >
              common.name
              <span
                aria-hidden={true}
                className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="on"
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="name-field"
                disabled={false}
                name="name"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-dhh0px"
                >
                  <span>
                    common.name
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            style={
              Object {
                "marginTop": 30,
              }
            }
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
            >
              common.email
              <span
                aria-hidden={true}
                className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="on"
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="email-field"
                disabled={false}
                name="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-dhh0px"
                >
                  <span>
                    common.email
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            style={
              Object {
                "marginTop": 30,
              }
            }
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
            >
              common.password
              <span
                aria-hidden={true}
                className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="password-field"
                disabled={false}
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="password"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-dhh0px"
                >
                  <span>
                    common.password
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            style={
              Object {
                "marginTop": 30,
              }
            }
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
            >
              common.repeatPassword
              <span
                aria-hidden={true}
                className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="repeat-password-field"
                disabled={false}
                name="repeatPassword"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="password"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-dhh0px"
                >
                  <span>
                    common.repeatPassword
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="makeStyles-buttonContainer-10"
          >
            <button
              className="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root css-utm9vu-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
              data-testid="submit-button"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              registration.signIn
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      className="makeStyles-loginFooter-9"
    >
      <div
        className="makeStyles-container-2"
      >
        registration.haveAnAccount
         
        <span
          className="makeStyles-link-1"
          id="sign-in-link"
          onClick={[Function]}
        >
          registration.signIn
        </span>
      </div>
    </div>
  </div>
</div>
`;
